app/src/main/java/org/literacyapp/authentication/TrainingHelper.java

```diff
@@ -171,4 +189,70 @@ private void deleteStudentImagesRecursive(StudentImage studentImage, String reas
+            }
+            if (archiveClassifierFiles()){
+                Log.i(getClass().getName(), "Classifier files have been archived.");
+            } else {
+                Log.e(getClass().getName(), "Failed to archive classifier files.");
+            }
+            Log.i(getClass().getName(), "Classifier training has started.");
+            svm.train("-t 0 ");
+            if (checkClassifierTrainingResult()){
+                for (StudentImage studentImage : studentImages){
+                    StudentImageCollectionEvent studentImageCollectionEvent = studentImage.getStudentImageCollectionEvent();
+                    if (!studentImageCollectionEvent.getSvmTrainingExecuted()){
+                        // Create new Student
+                        Student student = new Student();
+                        student.setUniqueId(StudentHelper.generateNextUniqueId(context, studentDao));
+                        student.setAvatar(studentImage);
+                        student.setTimeCreated(Calendar.getInstance());
+                        studentDao.insert(student);
+                        Log.i(getClass().getName(), "Student with Id " + student.getId() + " and uniqueId " + student.getUniqueId() + " has been created.");
+                        // Add Student to StudentImageCollectionEvent
+                        studentImageCollectionEvent.setStudent(student);
+                        studentImageCollectionEvent.setSvmTrainingExecuted(true);
+                        studentImageCollectionEventDao.update(studentImageCollectionEvent);
+                        Log.i(getClass().getName(), "StudentImageCollectionEvent with Id " + studentImageCollectionEvent.getId() + " has been trained in classifier");
+                    }
+                }
+                Log.i(getClass().getName(), "Classifier training has finished succuessfully.");
+            } else {
+                Log.e(getClass().getName(), "Classifier training has failed.");
+            }
+        }
+    }
+
+    private synchronized boolean archiveClassifierFiles(){
+        boolean success = true;
+        svmArchiveFolderWithTimestamp = new File(AiHelper.getSvmArchiveDirectory().getAbsolutePath(), Long.toString(new Date().getTime()));
+        Log.i(getClass().getName(), "Create archive directory " + svmArchiveFolderWithTimestamp.getAbsolutePath());
+        svmArchiveFolderWithTimestamp.mkdir();
+        success = success && svmTrainingFile.renameTo(new File(svmArchiveFolderWithTimestamp, svmTrainingFile.getName()));
+        success = success && svmTrainingModelFile.renameTo(new File(svmArchiveFolderWithTimestamp, svmTrainingModelFile.getName()));
+        if (!success){
+            svmArchiveFolderWithTimestamp.delete();
+        }
+        return success;
+    }
+
+    private synchronized boolean checkClassifierTrainingResult(){
```

Comment:
What does this check do, and why? Can you add a little bit of documentation to the method?